crates/router/src/connector/adyen/transformers.rs

```diff
@@ -73,6 +73,24 @@ struct AdyenBrowserInfo {
     java_enabled: bool,
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum AdyenStatus {
+    Authorised,
+    Refused,
+    Cancelled,
+    RedirectShopper,
+}
+
+impl From<AdyenStatus> for storage_enums::AttemptStatus {
+    fn from(item: AdyenStatus) -> Self {
+        match item {
+            AdyenStatus::Authorised => Self::Charged,
+            AdyenStatus::Refused | AdyenStatus::Cancelled => Self::Failure,
```

Comment:
`Cancelled` status should be mapped to `Self::Voided`. Correct me if I am wrong @SanchithHegde .

Reply:
You're right.